Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -619,7 +625,14 @@ static OSStatus sampleCallback(CMBufferRef buffer, void* refcon)
         if (m_decompressionSession)
             return;
 
-        m_decompressionSession = WebCoreDecompressionSession::createOpenGL();
+#if HAVE(RECOMMENDED_PIXEL_ATTRIBUTES_API)
+        if (m_renderer) {
+            if ([m_renderer.get() respondsToSelector:@selector(recommendedPixelBufferAttributes)])
+                m_decompressionSession = WebCoreDecompressionSession::create([m_renderer recommendedPixelBufferAttributes]);
```

Comment:
Can `recommendedPixelBufferAttributes` change during the lifetime of the renderer?
Should we account for that?

Reply:
it can't happen with the way the API works today. There's no KVM/observer possible etc. Right now it's a static set.